segment/segment.go

```diff
@@ -24,12 +24,14 @@ import (
 // larger frame spanning multiple segments (multi-segment part). It can also be optionally compressed with Lz4.
 // Segments were introduced in protocol v5 to improve compression efficiency (especially for small messages) and to
 // introduce error detection in the native protocol.
+// +k8s:deepcopy-gen=true
 type Segment struct {
```

Comment:
Types in this package did not have Clone methods so far, but for the sake of completeness, I'm generating methods for them as well.